examples/pytorch/model_zoo/chem/generative_models/dgmg/README.md

```diff
@@ -0,0 +1,104 @@
+# Learning Deep Generative Models of Graphs (DGMG)
+
+Yujia Li, Oriol Vinyals, Chris Dyer, Razvan Pascanu, and Peter Battaglia. 
+Learning Deep Generative Models of Graphs. *arXiv preprint arXiv:1803.03324*, 2018.
+
+DGMG generates graphs by progressively adding nodes and edges as below:
+![](https://user-images.githubusercontent.com/19576924/48605003-7f11e900-e9b6-11e8-8880-87362348e154.png)
+
+For molecules, the nodes are atoms and the edges are bonds.
+
+## Dataset
+
+### ChEMBL
+
+The authors use the [ChEMBL database](https://www.ebi.ac.uk/chembl/). Since they 
+did not release the code, we use a subset from [Olivecrona et al.](https://github.com/MarcusOlivecrona/REINVENT), 
+another work on generative modeling. 
+
+The authors restrict their dataset to molecules with at most 20 heavy atoms, and used a training/validation
+split of 130, 830/26, 166 examples each. We use the same split but need to relax 20 to 23 as we are using
+a different subset.
+
+### ZINC
+
+After the pre-processing, we are left with 232464 molecules for training and 5000 molecules for validation.
+
+## Usage
+
+### Using new datasets
+
+If you want to train with new datasets, you can proceed as follows:
+1. Get the list of atom types and bond types from a list of molecule SMILES. 
+You can use `get_atom_and_bond_types` at `L370, utils.py`.
+2. Since DGMG has some limitation in generating molecules (E.g. it ignores things like protonation and chirality), we
+need to standardize our dataset to avoid false novel molecules. You can preprocess your dataset with 
+`preprocess_dataset` at `L638, utils.py`. 
+3. Modify the dataset class `MoleculeDataset._setup` at `L753, utils.py`.
```

Comment:
It's not practical for user to use their own dataset under current interface. At least I still don't have a clear idea on how to customize after 15-min reading. Also it's not a good practice to ask user to rewrite a function which modifies `self`.

Reply:
Agree. I've made an update and see if you feel more comfortable now.